### ISSUE TITLE
Bug with training loop

### ISSUE THREAD

**Issue by OliverTryding** [OliverTryding/-RE-POLYGCL#1]
The loss is not improving, but is instead a constant value.
